design/CommandSubmission.md

```diff
@@ -0,0 +1,23 @@
+# Command Submission
+
+Users issue rendering and compute commands (such as resource bindings, draw calls, etc) via command buffers. The concept of `WebGPUCommandBuffer` matches the native graphics APIs. Those command buffers go through the following stages in their life cycle. It starts with creating a new `WebGPUCommandBuffer` from a `WebGPUCommandQueue` instance. From this point, the command buffer is considered to be in "recording" state.
+
+Commands can be encoded independently of the `WebGPUDevice`, or anything currently happening on GPU. The recording is CPU-only operation, and multiple command buffers can be recorded independently on web workers. (TODO: disallow recording multiple command buffers on the same thread/web worker?). Recording usually consists of a number of passes, be it render or compute, with occasional copy operations inserted between them.
```

Comment:
Not sure what "independently" means. Do you mean independently in time (aka it can be done while the GPU is busy) or device-independently (aka commands are represented in a device-independent byte code and then further compiled to the specific device later)?